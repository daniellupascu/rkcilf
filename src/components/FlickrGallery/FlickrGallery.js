import React from "react";
import PropTypes from "prop-types";

import {
  FlickrGalleryContainer,
  FlickrImageContainer,
  FlickrImage,
} from "./FlickrGalleryStyled";

const FlickrGallery = ({ images }) => {
  return (
    <FlickrGalleryContainer>
      {images.map((image, i) => (
        <FlickrImageContainer key={i}>
          <FlickrImage src={image.src} />
        </FlickrImageContainer>
      ))}
    </FlickrGalleryContainer>
  );
};

FlickrGallery.propTypes = {
  images: PropTypes.arrayOf(PropTypes.object),
};

export default FlickrGallery;
