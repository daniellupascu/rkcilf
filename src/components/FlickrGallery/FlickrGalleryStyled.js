import styled from "styled-components/macro";

export const FlickrGalleryContainer = styled.div`
  margin: 20px;
  position: relative;

  display: grid;
  grid-template-columns: repeat(4, minmax(150px, 1fr));
  grid-template-rows: repeat(auto-fill, minmax(200px, 1fr))
  grid-gap: 5px;
`;

export const FlickrImageContainer = styled.figure``;

export const FlickrImage = styled.img`
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: 10px;

  &:hover {
    transform: scale(1.5);
    transition: transform 0.2s;
    box-shaddow: 0px 24px 38px 3px rgba(0, 0, 0, 0.5),
      0px 9px 46px 8px rgba(0, 0, 0, 0.5), 0px 11px 15px -7px rgba(0, 0, 0, 0.5);
  }
`;
