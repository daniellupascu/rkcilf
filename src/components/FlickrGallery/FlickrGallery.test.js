import React from "react";
import renderer from "react-test-renderer";
import FlickrGallery from "./FlickrGallery";

describe("Test Flickr Gallery Component", () => {
  it("renders correctly", () => {
    const images = [
      {
        src: "url.com",
        title: { _text: "random Image" },
        author: { name: { _text: "Random User" } },
      },
      {
        src: "url.dk",
        title: { _text: "random Image" },
        author: { name: { _text: "Random User" } },
      },
    ];
    const tree = renderer.create(<FlickrGallery images={images} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
