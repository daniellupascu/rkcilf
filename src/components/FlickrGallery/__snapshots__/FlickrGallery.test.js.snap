// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Flickr Gallery Component renders correctly 1`] = `
<div
  className="FlickrGalleryStyled__FlickrGalleryContainer-sc-3z0gpy-0 kwOdIo"
>
  <figure
    className="FlickrGalleryStyled__FlickrImageContainer-sc-3z0gpy-1 kxaNwV"
  >
    <img
      className="FlickrGalleryStyled__FlickrImage-sc-3z0gpy-2 dxtjqM"
      src="url.com"
    />
    <div
      className="FlickrGalleryStyled__ImageInfo-sc-3z0gpy-3 bHTbjt"
    >
      <div
        className="FlickrGalleryStyled__ImageDetail-sc-3z0gpy-4 fhnvNK"
      >
        random Image
      </div>
      <div
        className="FlickrGalleryStyled__ImageDetail-sc-3z0gpy-4 fhnvNK"
      >
        credits: 
        Random User
      </div>
    </div>
  </figure>
  <figure
    className="FlickrGalleryStyled__FlickrImageContainer-sc-3z0gpy-1 kxaNwV"
  >
    <img
      className="FlickrGalleryStyled__FlickrImage-sc-3z0gpy-2 dxtjqM"
      src="url.dk"
    />
    <div
      className="FlickrGalleryStyled__ImageInfo-sc-3z0gpy-3 bHTbjt"
    >
      <div
        className="FlickrGalleryStyled__ImageDetail-sc-3z0gpy-4 fhnvNK"
      >
        random Image
      </div>
      <div
        className="FlickrGalleryStyled__ImageDetail-sc-3z0gpy-4 fhnvNK"
      >
        credits: 
        Random User
      </div>
    </div>
  </figure>
</div>
`;
